plugins/woocommerce/client/blocks/assets/js/blocks/product-filters/inner-blocks/attribute-filter/inspector.tsx

```diff
@@ -73,160 +60,235 @@ export const Inspector = ( {
-						isBlock
-						value={ queryType }
-						onChange={ ( value: BlockAttributes[ 'queryType' ] ) =>
-							setAttributes( { queryType: value } )
+						hasValue={ () =>
+							queryType !== metadata.attributes.queryType.default
 						}
-						__nextHasNoMarginBottom
-						__next40pxDefaultSize
-						style={ { width: '100%' } }
-						help={
-							queryType === 'and'
-								? createInterpolateElement(
-										__(
-											'Show results for <b>all</b> selected attributes. Displayed products must contain <b>all of them</b> to appear in the results.',
-											'woocommerce'
-										),
-										{
-											b: <strong />,
-										}
-								  )
-								: __(
-										'Show results for any of the attributes selected (displayed products don’t have to have them all).',
-										'woocommerce'
-								  )
+						onDeselect={ () =>
+							setAttributes( {
+								queryType:
+									metadata.attributes.queryType.default,
+							} )
 						}
 					>
-						<ToggleGroupControlOption
-							label={ __( 'Any', 'woocommerce' ) }
-							value="or"
-						/>
-						<ToggleGroupControlOption
-							label={ __( 'All', 'woocommerce' ) }
-							value="and"
-						/>
-					</ToggleGroupControl>
-					<ToggleGroupControl
-						value={ displayStyle }
-						isBlock
-						__nextHasNoMarginBottom
-						__next40pxDefaultSize
-						onChange={ (
-							value: BlockAttributes[ 'displayStyle' ]
-						) => {
-							if ( ! filterBlock ) return;
-							const currentStyleBlock = getInnerBlockByName(
-								filterBlock,
-								displayStyle
-							);
-
-							if ( currentStyleBlock ) {
-								setDisplayStyleBlocksAttributes( {
-									...displayStyleBlocksAttributes,
-									[ displayStyle ]:
-										currentStyleBlock.attributes,
-								} );
-								replaceBlock(
-									currentStyleBlock.clientId,
-									createBlock(
-										value,
-										displayStyleBlocksAttributes[ value ] ||
-											{}
-									)
-								);
-							} else {
-								insertBlock(
-									createBlock( value ),
-									filterBlock.innerBlocks.length,
-									filterBlock.clientId,
-									false
-								);
+						<ToggleGroupControl
+							label={ __( 'Logic', 'woocommerce' ) }
+							isBlock
+							value={ queryType }
+							onChange={ (
+								value: BlockAttributes[ 'queryType' ]
+							) => setAttributes( { queryType: value } ) }
+							__nextHasNoMarginBottom
+							__next40pxDefaultSize
+							style={ { width: '100%' } }
+							help={
+								queryType === 'and'
+									? createInterpolateElement(
+											__(
+												'Show results for <b>all</b> selected attributes. Displayed products must contain <b>all of them</b> to appear in the results.',
+												'woocommerce'
+											),
+											{
+												b: <strong />,
+											}
+									  )
+									: __(
+											"Show results for any of the attributes selected (displayed products don't have to have them all).",
```

Comment:
I know it's not introduced in this PR, but this text sounds a bit unnatural to me. I would replace it with something along these lines:

```suggestion
											"Display products that match any of the selected attributes (they don't need to match all).",
```